src/client/runner/automation/playback/scroll.js

```diff
@@ -1,364 +1,185 @@
-            targetHeight        = isHtmlElement ? target.clientHeight : targetRect.height;
-            offsetX             = actionOptions && typeof actionOptions.offsetX !==
-                                                   'undefined' ? actionOptions.offsetX : Math.round(targetWidth / 2);
-            offsetY             = actionOptions && typeof actionOptions.offsetY !==
-                                                   'undefined' ? actionOptions.offsetY : Math.round(targetHeight / 2);
-            maxLeftScrollMargin = Math.min(MAX_SCROLL_MARGIN, Math.floor(targetWidth / 2));
-            maxTopScrollMargin  = Math.min(MAX_SCROLL_MARGIN, Math.floor(targetHeight / 2));
-            ownScroll           = styleUtils.getElementScroll(to);
-
-            if (!styleUtils.hasScroll(to)) {
-                scrollElementCallback();
-                return;
-            }
+            currentOffsetX += childDimensions.left - parentDimensions.left + parentDimensions.border.left;
+            currentOffsetY += childDimensions.top - parentDimensions.top + parentDimensions.border.top;
+
+            currentChild = parents[i];
+        }
 
-            if (offsetX >= ownScroll.left + targetWidth)
-                newOwnScrollLeft = offsetX - maxLeftScrollMargin;
-            else if (offsetX <= ownScroll.left)
-                newOwnScrollLeft = offsetX - maxLeftScrollMargin;
-
-            if (offsetY >= ownScroll.top + targetHeight)
-                newOwnScrollTop = offsetY - maxTopScrollMargin;
-            else if (offsetY <= ownScroll.top)
-                newOwnScrollTop = offsetY - maxTopScrollMargin;
-
-            async.series({
-                scrollBody: function (callback) {
-                    if (isHtmlElement)
-                        scrollElement(target.getElementsByTagName('body')[0], newOwnScrollLeft, newOwnScrollTop, callback);
-                    else
-                        callback();
-                },
-                ownScroll:  function (callback) {
-                    scrollElement(target, newOwnScrollLeft, newOwnScrollTop, callback);
-                },
-                callback:   function () {
-                    scrollElementCallback();
-                }
+        if (window.top !== window.self) {
+            return sendRequestToParentFrame({
+                cmd:     'scroll',
+                offsetX: currentOffsetX,
+                offsetY: currentOffsetY
             });
-        },
+        }
 
-        scrollElementParents: function (scrollParentsCallback) {
-            scrollableParents = isDomElement ? styleUtils.getScrollableParents(to, currentDocument) : [document.documentElement];
+        return Promise.resolve();
+    }
 
-            if (!scrollableParents.length) {
-                scrollParentsCallback();
-                return;
-            }
+    run () {
+        Scroll._scrollElementToPoint(this.element, { x: this.offsetX, y: this.offsetY });
```

Comment:
Why they are static? If you'll make them instance methods there will be no need to pass this arguments around.